### PR TITLE
Refactor TaskRun to functional component

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

#2000 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/main/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by AlanGreene** [comment]
Seems like something wrong with prow, the build and unit test jobs actually completed successfully, and the integration tests job just stopped with no error.

/retest

File: src/containers/TaskRun/TaskRun.js [hunk 1]
```diff
@@ -150,215 +161,178 @@ export /* istanbul ignore next */ class TaskRunContainer extends Component {
...
 
     const browserURL = match.url.concat(`?${queryParams.toString()}`);
     history.push(browserURL);
-  };
+  }
 
-  showRerunNotification = value => {
-    this.setState({ showRerunNotification: value });
-  };
+  function showRerunNotification(value) {
```
Comment (briangleeson) [inline src/containers/TaskRun/TaskRun.js:201]: Is the `showRerunNotification` function a bit redundant now? Could just call `setRerunNotification` directly?

Comment (AlanGreene) [reply src/containers/TaskRun/TaskRun.js]: Absolutely, good call! Updated to call `setRerunNotification` directly

**Pull request merged**